### PR TITLE
Add tests for github

### PR DESCRIPTION
**Pull request by VenelinMartinov**
https://github.com/pulumi/pulumi-terraform-bridge/issues/1849

This PR adds tests for the following resources:

index_repositoryautolinkreference
index_repositorywebhook

### CODE REVIEW

File: provider/provider_program_test.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+		t.Skipf("Skipping in testing.Short() mode, assuming this is a CI run without credentials")
+	}
+	cwd, err := os.Getwd()
+	require.NoError(t, err)
+	if baselineVersion == "" {
+		baselineVersion = defaultBaselineVersion
+	}
+	test := pulumitest.NewPulumiTest(t, dir,
+		opttest.DownloadProviderVersion(providerName, baselineVersion),
+		opttest.LocalProviderPath(providerName, filepath.Join(cwd, "..", "bin")),
```
Comment (guineveresaenger) [inline provider/provider_program_test.go:78]: You'll want to set `github:token` and `github:owner` to `GITHUB_TESTING_TOKEN` and `GITHUB_ORG`, respectively. Then CI should pass.

See how integration tests are set up: 
https://github.com/pulumi/pulumi-github/blob/master/examples/examples_test.go#L15

**Review by guineveresaenger — CHANGES_REQUESTED** [review]
I'm not 💯 sure if our [test org](https://github.com/moolumi) token is correctly scoped for these two specific resources but getting those env vars set in the programtest file should be the way to go.

File: provider/go.mod [hunk 2]
```diff
@@ -23,6 +23,8 @@ require (
 //	        github.com/hashicorp/terraform-exec/tfinstall: module github.com/hashicorp/terraform-exec@latest found (v0.20.0), but does not contain package github.com/hashicorp/terraform-exec/tfinstall
 replace github.com/hashicorp/terraform-exec => github.com/hashicorp/terraform-exec v0.15.0
 
+require github.com/stretchr/testify v1.9.0
```
Comment (guineveresaenger) [inline provider/go.mod:26]: nit: can this be grouped with the other `require`s?

**Review by guineveresaenger — APPROVED** [review]

**Comment by VenelinMartinov** [comment]
blocked on https://github.com/pulumi/pulumi-github/issues/636

**Pull request closed without merging**